### PR TITLE
Avoid converting literal node input type by tool definition when load tool

### PR DESCRIPTION
**Pull request by huaiyan**
# Description

Avoid converting literal node input type by tool definition.
UX/extension should assign literal node input by correct format in yaml.
Executor could skip conversion.
That will be better for customer's debugging, since all tools' could receive original content in yaml.

Warning log:
image

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [X] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [X] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/contracts/tool.py [hunk 1]
```diff
@@ -135,6 +135,21 @@ def parse(self, v: Any) -> Any:  # noqa: C901
...
+            return float
+        if self == ValueType.BOOL:
+            return bool
+        if self == ValueType.STRING:
+            return str
+        if self == ValueType.LIST:
+            return list
+        if self == ValueType.OBJECT:
+            return dict
+        return None
```
Comment (thy09) [inline src/promptflow/promptflow/contracts/tool.py:151]: Image?

File: src/promptflow/promptflow/contracts/tool.py [hunk 2]
```diff
@@ -135,6 +135,21 @@ def parse(self, v: Any) -> Any:  # noqa: C901
         # TODO: parse other types
         return v
 
+    def get_class(self):
```
Comment (thy09) [inline src/promptflow/promptflow/contracts/tool.py:138]: We need to have a docstring if we want to expose this.

Comment (thy09) [inline src/promptflow/promptflow/contracts/tool.py:138]: get_python_type?

File: src/promptflow/promptflow/executor/_tool_resolver.py [hunk 3]
```diff
@@ -106,7 +107,12 @@ def _convert_node_literal_input_types(self, node: Node, tool: Tool, module: type
                 updated_inputs[k].value = create_image(v.value)
             elif isinstance(value_type, ValueType):
                 try:
-                    updated_inputs[k].value = value_type.parse(v.value)
+                    updated_inputs[k].value = v.value
+                    # Don't convert value type for literal node input.
```
Comment (thy09) [inline src/promptflow/promptflow/executor/_tool_resolver.py:111]: Then I think we don't need the following converting error?

File: src/promptflow/promptflow/executor/_tool_resolver.py [hunk 4]
```diff
@@ -106,7 +107,12 @@ def _convert_node_literal_input_types(self, node: Node, tool: Tool, module: type
...
+                    # We don't covert flow input/ node reference type by tool definition.
+                    # So, don't have reason to convert literal node input type.
+                    if not isinstance(updated_inputs[k].value, value_type.get_class()):
+                        logger.warning(f"Literal node input {k} is not type {value_type}, received type is {type(v)}.")
                     updated_inputs[k].value = load_multimedia_data_recursively(updated_inputs[k].value)
```
Comment (thy09) [inline src/promptflow/promptflow/executor/_tool_resolver.py:116]: I guess we can load multimedia before the type check?

File: src/promptflow/promptflow/contracts/tool.py (hunk 1 above)
Comment (huaiyan) [reply src/promptflow/promptflow/contracts/tool.py:151]: We still need to convert for image type.
Will add comment to explain.

File: src/promptflow/promptflow/executor/_tool_resolver.py (hunk 4 above)
Comment (huaiyan) [reply src/promptflow/promptflow/executor/_tool_resolver.py:118]: No, after loading multimedia, we may get `Image` for `Dict`

File: src/promptflow/promptflow/executor/_tool_resolver.py (hunk 3 above)
Comment (huaiyan) [reply src/promptflow/promptflow/executor/_tool_resolver.py:113]: No, we still need to handle composite objects. Such as list of images.
But it only happen with specific image format.

**Pull request closed without merging**